### PR TITLE
Validate preconditions

### PR DESCRIPTION
**Pull request by edkajimenez**
## Summary
Add Guava to config
Validate not null preconditions on models setters

## Trello Card
https://trello.com/c/RlaXq1yF/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -65,6 +66,7 @@ public String getTitle() {
     }
 
     public void setTitle(String title) {
+        Preconditions.checkNotNull(title);
```
Comment (zeminlu) [inline src/main/java/wolox/training/models/Book.java]: You could also check for the string not being empty.
You  can do `Preconditions.checkArgument(title != null && !title.isEmpty());`
Also, add the check for the setPages setter, so it checks it's > 0.

File: src/main/java/wolox/training/models/User.java [hunk 2]
```diff
@@ -51,6 +52,7 @@ public String getUserName() {
     }
 
     public void setUserName(String userName) {
+        Preconditions.checkNotNull(userName);
```
Comment (zeminlu) [inline src/main/java/wolox/training/models/User.java]: Same as comments in the othe model.

File: src/main/java/wolox/training/models/Book.java [hunk 3]
```diff
@@ -81,6 +84,7 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
+        Preconditions.checkNotNull(genre);
```
Comment (zeminlu) [inline src/main/java/wolox/training/models/Book.java]: This one is nullable. I would only check that , IF it's not null, then check that it's not empty.

**Review by zeminlu — APPROVED** [review]
Looks good! Check my comments.

**Review by zeminlu — CHANGES_REQUESTED** [review]
I accidentaly approved it, check my comments.

File: src/main/java/wolox/training/models/Book.java (hunk 1 above)
Comment (edkajimenez) [reply src/main/java/wolox/training/models/Book.java]: Got it!

**Review by zeminlu — APPROVED** [review]

**Pull request merged**